core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -261,17 +386,17 @@ public DeleteWriteBuilder rowSchema(org.apache.iceberg.Schema newRowSchema) {
       return this;
     }
 
-    public DeleteWriteBuilder withSpec(PartitionSpec newSpec) {
+    public DeleteWriteBuilder spec(PartitionSpec newSpec) {
```

Comment:
I think naming in this builder is inconsistent as only some methods use `withPrefix`. We don't have that prefix in the appender builder too. I think it should be fine to rename now as the delta code path is evolving.